latest openscad on tumbleweed does not work with wayland

**Describe the bug**

As you know I build openscad for Tumbleweed rpm (https://build.opensuse.org/project/show/home:joba1:openscad)
It does not work with wayland. It works only if started like this (X11 compatibility):
```
QT_QPA_PLATFORM=xcb openscad
```
or like this:
```
ssh -Y job6 openscad
```

If I just run `openscad` I see these messages in the openscad Console:
```
ERROR: Framebuffer Objects not supported
ERROR: MouseSelector: Failed to create framebuffer; disabling mouse selection.
```

And I see these error messages in the terminal 
```
/home/abuild/rpmbuild/BUILD/openscad-2026.xx.xx+git.20260224T222412_ae3a780d-build/openscad-2026.xx.xx+git.20260224T222412~ae3a780d/src/glview/ShaderUtils.cc: OpenGL vertex shader Error:
0(2) : error C0201: unsupported version 120
0(3) : error C7532: global type vec3 requires "#version 100" or later
0(7) : error C7532: global type float requires "#version 100" or later
0(8) : error C7532: global type vec4 requires "#version 100" or later
0(11) : error C7532: global variable gl_Position requires "#version 100" or later
0(11) : error C7614: OpenGL/ES doesn't allow use of reserved word gl_ModelViewProjectionMatrix
0(11) : error C7532: global variable gl_ModelViewProjectionMatrix requires "#version 100" or later
0(11) : error C0000: ... or #extension GL_ARB_compatibility : enable
0(11) : error C7614: OpenGL/ES doesn't allow use of reserved word gl_Vertex
0(11) : error C7532: global variable gl_Vertex requires "#version 100" or later
0(11) : error C0000: ... or #extension GL_ARB_compatibility : enable
0(14) : error C7614: OpenGL/ES doesn't allow use of reserved word gl_NormalMatrix
0(14) : error C7532: global variable gl_NormalMatrix requires "#version 100" or later
0(14) 
```

**To Reproduce**
Steps to reproduce the behavior:
1. install my rpm
2. run openscad
3. load any model file and render
4. See errors (and no rendered model)

**Expected behavior**
No error messages, visible model after rendering

**Code reproducing the issue**
n/a

**Screenshots**
n/a

**Environment and Version info (please complete the following information):**
 - OS: Tumbleweed
 - System: AMD PC 64-bit
 - OpenSCAD Version: 2026.02.26 from https://download.opensuse.org/repositories/home:/joba1:/openscad/openSUSE_Tumbleweed/

**Library & Graphics card information**
joachim@job6:~> openscad --info
QObject::startTimer: Timers can only be used with threads started with QThread
OpenSCAD Version: 2026.02.26
System information: Linux 6.17.7-1-default #1 SMP PREEMPT_DYNAMIC Sun Nov  2 16:18:06 UTC 2025 (5d304cd) x86_64 openSUSE Tumbleweed 16 CPUs 61.95 GB RAM
User Agent: OpenSCAD/2026.02.26 (Linux x86_64; openSUSE Tumbleweed)
Compiler: GCC "15.2.1 20260202" 64bit
MinGW build: No
Debug build: No
Boost version: 1_90
Eigen version: 3.5.0
CGAL version, kernels: 6.1, Cartesian<Gmpq>, Extended_cartesian<Gmpq>
OpenCSG version: OpenCSG 1.8.1
Clipper2 version: 2.0.1
Manifold version: 3.3.2
Qt version: 6.10.0
QScintilla version: 2.14.1
InputDrivers: 
GLib version: 2.86.4
lodepng version: 20260102
libzip version: 1.11.4
fontconfig version: 2.16.0
freetype version: 2.14.1
harfbuzz version: 12.3.2
cairo version: 1.18.4
lib3mf version: 2.4.1
Features: roof, input-driver-dbus, lazy-union, vertex-object-renderers-indexing, textmetrics, import-function, object-function, predictible-output, vector-swizzle, discretization-by-error
Application Path: /usr/bin
Documents Path: /home/joachim/.local/share
User Documents Path: /home/joachim/Nextcloud/Documents
Resource Path: /usr/share/openscad
User Library Path: /home/joachim/.local/share/OpenSCAD/libraries
User Examples Path: /home/joachim/.local/share/OpenSCAD/examples
User Config Path: /home/joachim/.config/OpenSCAD
Backup Path: /home/joachim/.local/share/OpenSCAD/backups
OPENSCADPATH: <not set>
OpenSCAD library path:
  /home/joachim/.local/share/OpenSCAD/libraries
  /usr/share/openscad/libraries

OPENSCAD_FONT_PATH: <not set>
OpenSCAD font path:
  /usr/X11R6/lib/X11/fonts
  /opt/kde3/share/fonts
  /home/joachim/.local/share/fonts
  /home/joachim/.local/share/flatpak/exports/share/fonts
  /var/lib/flatpak/exports/share/fonts
  /usr/local/share/fonts
  /usr/share/fonts
  /home/joachim/.fonts
  /usr/X11/lib/X11/fonts
  /System/Library/Fonts
  /Library/Fonts
  /home/joachim/Library/Fonts
  /usr/share/fonts/100dpi
  /usr/share/fonts/75dpi
  /usr/share/fonts/Adwaita
  /usr/share/fonts/Type1
  /usr/share/fonts/cyrillic
  /usr/share/fonts/encodings
  /usr/share/fonts/ghostscript
  /usr/share/fonts/misc
  /usr/share/fonts/truetype
  /usr/share/fonts/xscreensaver
  /usr/share/fonts/encodings/large

GL context creator: EGL (new)
EGL version: 1.5
PNG generator: lodepng

GLAD version: 2.0.4
OpenGL Version: 4.6.0 NVIDIA 580.105.08
GL Renderer: NVIDIA GeForce RTX 5060 Ti/PCIe/SSE2
GL Vendor: NVIDIA Corporation
RGBA(8888), depth(24), stencil(8)
GL_ARB_framebuffer_object: yes
GL_EXT_framebuffer_object: yes
GL_EXT_packed_depth_stencil: yes

**Additional context**
Do I need to add libraries to the build? Or use special complie/link flags? Or is this known/normal?
AFAIK X11 compatibility layer is going to be removed soon (I use plasma, but I think gnome already did that)

```
joachim@job6:~> inxi -G
Graphics:
  Device-1: NVIDIA GB206 [GeForce RTX 5060 Ti] driver: nvidia v: 580.105.08
  Device-2: Advanced Micro Devices [AMD/ATI] Raphael driver: amdgpu
    v: kernel
  Display: wayland server: X.org v: 1.21.1.15 with: Xwayland v: 24.1.8
    compositor: kwin_wayland driver: X: loaded: modesetting,nvidia
    unloaded: vesa dri: radeonsi gpu: nv_platform,nvidia,nvidia-nvswitch
    resolution: 3440x1440~75Hz
  API: EGL v: 1.5 drivers: nvidia,radeonsi,swrast
    platforms: gbm,wayland,x11,surfaceless,device
  API: OpenGL v: 4.6.0 compat-v: 4.5 vendor: nvidia mesa v: 580.105.08
    renderer: NVIDIA GeForce RTX 5060 Ti/PCIe/SSE2
  API: Vulkan v: 1.4.328 drivers: nvidia,llvmpipe surfaces: N/A
  Info: Tools: api: clinfo, eglinfo, glxinfo, vulkaninfo
    de: kscreen-console,kscreen-doctor gpu: nvidia-smi wl: wayland-info
    x11: xdpyinfo, xprop, xrandr
```

<!-- If you want to assign this issue to yourself, write /assign-me on its own line -->

--- a/src/gui/QGLView.cc
+++ b/src/gui/QGLView.cc
@@ -42,6 +42,7 @@
 
 #include <QImage>
 #include <QOpenGLWidget>
+#include <QSurfaceFormat>
 #include <QWidget>
 #include <iostream>
 #include <QApplication>
@@ -72,8 +73,23 @@
 #include "gui/qt-obsolete.h"
 #include "gui/Measurement.h"
 
+namespace {
+
+QSurfaceFormat compatibleWidgetFormat()
+{
+  auto format = QSurfaceFormat::defaultFormat();
+  format.setRenderableType(QSurfaceFormat::OpenGL);
+  format.setProfile(QSurfaceFormat::CompatibilityProfile);
+  if (format.depthBufferSize() < 24) format.setDepthBufferSize(24);
+  if (format.stencilBufferSize() < 8) format.setStencilBufferSize(8);
+  return format;
+}
+
+}  // namespace
+
 QGLView::QGLView(QWidget *parent) : QOpenGLWidget(parent)
 {
+  setFormat(compatibleWidgetFormat());
   init();
 }
 

--- a/src/openscad_gui.cc
+++ b/src/openscad_gui.cc
@@ -36,6 +36,7 @@
 #include <QIcon>
 #include <QObject>
 #include <QPalette>
+#include <QSurfaceFormat>
 #include <QStringList>
 #include <QStyleHints>
 #include <Qt>
@@ -110,6 +111,25 @@ bool isDarkMode()
 #endif  // QT_VERSION
 }
 
+void configureOpenGLContext()
+{
+#if defined(Q_OS_UNIX) && !defined(Q_OS_MACOS)
+  // OpenSCAD still relies on legacy OpenGL compatibility features and GLSL 1.20
+  // shaders, so a GLES context is not currently usable. On Wayland/EGL setups Qt
+  // can otherwise pick OpenGL ES by default.
+  if (qEnvironmentVariableIsEmpty("QT_OPENGL")) {
+    QCoreApplication::setAttribute(Qt::AA_UseDesktopOpenGL);
+  }
+
+  auto format = QSurfaceFormat::defaultFormat();
+  format.setRenderableType(QSurfaceFormat::OpenGL);
+  format.setProfile(QSurfaceFormat::CompatibilityProfile);
+  if (format.depthBufferSize() < 24) format.setDepthBufferSize(24);
+  if (format.stencilBufferSize() < 8) format.setStencilBufferSize(8);
+  QSurfaceFormat::setDefaultFormat(format);
+#endif
+}
+
 }  // namespace
 
 namespace {
@@ -178,6 +198,7 @@ void registerDefaultIcon(const QString&)
 int gui(std::vector<std::string>& inputFiles, const std::filesystem::path& original_path, int argc,
         char **argv, const std::string& gui_test, const bool reset_window_settings)
 {
+  configureOpenGLContext();
   OpenSCADApp app(argc, argv);
   QIcon::setThemeName(isDarkMode() ? "chokusen-dark" : "chokusen");
 

